Mathlib/Probability/Distributions/Exponential.lean

```diff
@@ -0,0 +1,281 @@
+  _ = ∫⁻ x in {x | x ≥ c} ∪ {x | x < c} , f x ∂ volume := by rw [← union]
+  _ = _ := by
+    apply lintegral_union this.measurableSet
+    rw [Set.disjoint_iff]; rintro x ⟨hxge : x ≥ _, hxlt : x < _⟩; linarith
+
+namespace ProbabilityTheory
+
+section ExponentialPdf
+
+/-- Define the PDF of the exponential distribution depending on its rate-/
+noncomputable
+def exponentialPdfReal (r x : ℝ) : ℝ :=
+  if 0 ≤ x then r*exp (-(r*x)) else 0
+
+local notation "E" => exponentialPdfReal
+
+/-- The PDF of the exponential Distribution on the extended real Numbers-/
+noncomputable
+def exponentialPdf (r x : ℝ) : ℝ≥0∞ :=
+  ENNReal.ofReal (E r x)
+
+lemma exponentialPdf_eq (r x : ℝ):
+    exponentialPdf r x = ENNReal.ofReal (if 0 ≤ x then r*exp (-(r*x)) else 0) := rfl
+
+lemma antiDeriv_expDeriv_pos' {r x : ℝ} (hr : 0 < r) :
+    HasDerivAt (fun a ↦ -1/r * exp (-(r * a))) (exp (-(r * x))) x := by
+  convert (((hasDerivAt_id x).const_mul (-r)).exp.const_mul (-1/r)) using 1 <;> field_simp
+
+/-- the Lebesgue-Integral of the exponential PDF over nonpositive Reals equals 0-/
+lemma lintegral_nonpos {x r : ℝ} (hx : x ≤ 0) :
+    ∫⁻ y in Iio x, exponentialPdf r y = ENNReal.ofReal 0 := by
+  rw [set_lintegral_congr_fun (g := fun _ ↦ 0) measurableSet_Iio]
+  · rw [lintegral_zero, ← ENNReal.ofReal_zero]
+  · refine ae_of_all _ ?_
+    intro a (ha : a < x)
+    simp only [exponentialPdf_eq, ge_iff_le, ENNReal.ofReal_eq_zero]
+    rw [if_neg]
+    linarith
+
+/-- The exponential pdf is measurable. -/
+lemma measurable_exponentialPdfReal (r : ℝ) :
+    Measurable (E r) := by
+  unfold exponentialPdfReal
+  refine Measurable.ite ?hp ((measurable_id'.const_mul r).neg.exp.const_mul r) ?hg
+  · refine MeasurableSet.of_compl ?hp.h
+    apply IsOpen.measurableSet; rw [compl_setOf_le]; exact isOpen_gt' 0
+  · exact measurable_const
+
+/-- The exponential Pdf is strongly measurable -/
+lemma stronglyMeasurable_exponentialPdfReal (r : ℝ) :
+    StronglyMeasurable (E r) :=
+  (measurable_exponentialPdfReal r).stronglyMeasurable
+
+/-- the exponential Pdf is positive for all positive reals-/
+lemma exponentialPdfReal_pos {x r : ℝ} {hr : 0 < r} (hx : 0 < x) :
+    0 < E r x := by
+  simp only [exponentialPdfReal, if_pos hx.le]
+  positivity
+
+/-- The exponential Pdf is nonnegative-/
+lemma exponentialPdfReal_nonneg {r : ℝ} (hr : 0 < r) :
+    ∀ x : ℝ, 0 ≤ E r x := by
+  unfold exponentialPdfReal
+  intro x
+  split_ifs <;> positivity
+
+/-- A negative exponential function is integrable on Intervals in R≥0 -/
+lemma exp_neg_integrableOn_Ioc {b x : ℝ} (hb : 0 < b) :
+    IntegrableOn (fun x ↦ rexp (-(b * x))) (Ioc 0 x) := by
+  simp only [neg_mul_eq_neg_mul]
+  exact (exp_neg_integrableOn_Ioi _ hb).mono_set Ioc_subset_Ioi_self
+
+lemma if_eval_pos {r : ℝ} : ∀ᵐ x : ℝ ∂ volume , x < 0 →
+    ENNReal.ofReal (if x ≥ 0 then r * rexp (-(r * x)) else 0) = 0 := by
+  apply ae_of_all
+  intro x hx
+  replace hx : ¬ 0 ≤ x := by exact not_le.mpr hx
+  simp [hx]
```

Comment:
Several things here:
- all lemmas in mathlib use `≤` instead of `≥` so it's better to use `0 ≤ x`
- the proof can be made very simple
- the name is not specific enough: this is not about any `if`, it's about the particular one in `exponentialPdf`
- this should probably be rewritten as a lemma about `exponentialPdf`
```suggestion
    ENNReal.ofReal (if 0 ≤ x then r * rexp (-(r * x)) else 0) = 0 :=
  ae_of_all _ (fun x hx ↦ by simp [not_le.mpr hx])
```